src/pages-and-resources/teams/TeamSetEditor.jsx

```diff
@@ -0,0 +1,193 @@
+
+/**
+ * Team sizes enum
+ * @enum
+ * @type {{MIN: number, MAX: number, DEFAULT: number}}
+ */
+export const TeamSizes = {
+  DEFAULT: 50,
+  MIN: 1,
+  MAX: 500,
+};
+
+/**
+ * Team types enum
+ * @enum
+ * @type {{PRIVATE_MANAGED: string, PUBLIC_MANAGED: string, OPEN: string}}
+ */
+export const TeamSetTypes = {
+  OPEN: 'open',
+  PUBLIC_MANAGED: 'public_managed',
+  PRIVATE_MANAGED: 'private_managed',
+};
+
+// Maps a team type to its corresponding intl message
+const TeamTypeNameMessage = {
+  [TeamSetTypes.OPEN]: messages.teamSetTypeOpen,
+  [TeamSetTypes.PUBLIC_MANAGED]: messages.teamSetTypePublicManaged,
+  [TeamSetTypes.PRIVATE_MANAGED]: messages.teamSetTypePrivateManaged,
+};
+
+function TeamSetEditor({
+  intl, teamSet, onDelete, onChange, onBlur, fieldNameCommonBase, errors, touched,
+}) {
+  const [isDeleting, setDeleting] = useState(false);
+  const initiateDeletion = () => setDeleting(true);
+  const cancelDeletion = () => setDeleting(false);
+
+  return (
+    <Collapsible
+      title={(
+        <div className="d-flex flex-column small flex-shrink-1">
+          <span>{intl.formatMessage(TeamTypeNameMessage[teamSet.type])}</span>
+          <span className="h4 text-truncate">{teamSet.name || '<new>'}</span>
+          <span className="text-truncate">{teamSet.description || '<new>'}</span>
+        </div>
+      )}
+      // If there is no id, this is a new team, so automatically open it
+      defaultOpen={teamSet.id === null}
+    >
+      <TransitionReplace>
+        {isDeleting
+          ? (
+            <div className="d-flex flex-column m-4" key="isDeleting">
+              <h3>{intl.formatMessage(messages.teamSetDeleteHeading)}</h3>
+              <p>{intl.formatMessage(messages.teamSetDeleteBody)}</p>
+              <div className="d-flex flex-row justify-content-end">
+                <Button variant="muted" size="sm" onClick={() => onDelete(teamSet)}>
+                  {intl.formatMessage(messages.delete)}
+                </Button>
+                <Button variant="muted" size="sm" onClick={cancelDeletion}>
+                  {intl.formatMessage(messages.cancel)}
+                </Button>
+              </div>
+            </div>
+          )
+          : (
+            <React.Fragment key="isConfiguring">
+              <Form.Group key="isConfiguring">
+                <Form.Control
+                  name={`${fieldNameCommonBase}.name`}
+                  floatingLabel={intl.formatMessage(messages.teamSetFormNameLabel)}
+                  defaultValue={teamSet.name}
+                  onChange={onChange}
+                  onBlur={onBlur}
+                />
+                <Form.Text>{intl.formatMessage(messages.teamSetFormNameHelp)}</Form.Text>
+                {(touched?.name && errors?.name) && (
+                <Form.Control.Feedback type="invalid" hasIcon={false}>
+                  {intl.formatMessage(messages.teamSetFormNameError)}
+                </Form.Control.Feedback>
+                )}
```

Comment:
for all the error messages I would simplify it by creating a generic component for example

```
function ErrorMessage({ errors, touched,  name }) {
  if (!errors[name] || !touched[name]) {
    return null;
  }
  return <p>{errors[name]}</p>;
}
```
and in this case, it can be modified as 

```
function ErrorMessage({ errors, touched,  name, key }) {
  if (!errors[name] || !touched[name]) {
    return null;
  }
  return <p>{intl.formatMessage(messages[key])}</p>;
}
```
and usage will be like 

`<ErrorMessage errors={errors} touched={touched}  name={name}  key='teamSetFormNameError' />`

but if we transfer error messages to yup validations it will get shorten like in the first example. and I will suggest shifting error messages with yup validations.